src/types/pysuper.rs

```diff
@@ -0,0 +1,26 @@
+use crate::ffi;
+use crate::type_object::PyTypeInfo;
+use crate::types::{PyTuple, PyType};
+use crate::{AsPyPointer, Py, PyAny, PyErr, PyResult, Python};
+
+/// Represents a Python `super` object.
+///
+/// This type is immutable.
+#[repr(transparent)]
+pub struct PySuper(PyAny);
+
+pyobject_native_type_core!(PySuper, ffi::PySuper_Type, #checkfunction=ffi::PyType_Check);
```

Comment:
I'm not convinced that `PyType_Check` is correct, that will check if objects are of type `type` ?

I think better to remove this and use the default implementation.

```suggestion
pyobject_native_type_core!(PySuper, ffi::PySuper_Type);
```

We could verify that this works correctly by writing a test which calls `PySuper::is_type_of(super_obj)` for some `super_obj: PySuper`.

Reply:
Thank you! Done!